code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -900,6 +900,29 @@ obj/item/assembly/radio_horn/receive_signal()
 
 			//consume material
 			craftingitem.change_stack_amount(-consumed)
+
+
+/datum/pipeshotrecipe/plasglass //was tricky figuring this out in a non-abusable way but we got there
+	thingsneeded = 2
+	result = /obj/item/ammo/bullets/pipeshot/plasglass
+	accepteditem = /obj/item/raw_material/shard
+
+	craftwith(obj/item/craftingitem, obj/item/frame, mob/user) //overwritten to change the typecheck to a material check
```

Comment:
rather than overwritting this individual craftwith, you can add a var `mat_id` to the abstract parent, and update the logic so that you check both `accepteditem` and `mat_id` pass, if `mat_id` is `nonnull`.